File: ios/NamiManager.m
//
//  NamiManager.m
//  RNNami
//
//  Copyright © 2020-2023 Nami ML Inc. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNNamiManager, NSObject)

RCT_EXTERN_METHOD(sdkConfigured:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup {
  return YES;
}

@end

File: ios/NamiManager.swift
//
//  NamiManager.swift
//  RNNami
//
//  Copyright © 2023 Nami ML INc.. All rights reserved.
//

import Foundation
import NamiApple
import React

@objc(RNNamiManager)
class RNNamiManager: RCTEventEmitter {
    public static var shared: RNNamiManager?

    override init() {
        super.init()
        RNNamiManager.shared = self
    }

    override func supportedEvents() -> [String]! {
        return []
    }

    @objc(sdkConfigured:rejecter:)
    func sdkConfigured(resolve: @escaping RCTPromiseResolveBlock, reject _: @escaping RCTPromiseRejectBlock) {
        let sdkConfigured = Nami.sdkConfigured()
        resolve(sdkConfigured)
    }
}
